### ISSUE TITLE
Search timeout if installed with 3 querynodes

### ISSUE THREAD

**Issue by del-zhenwu** [milvus-io/milvus#6417]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
Some requests get "Deadline Exceeded" during searching with locust, 100-users generated.

```
[2021-07-09 08:39:27,556] [    INFO] -  Name                                                          # reqs      # fails  |     Avg
     Min     Max  Median  |   req/s failures/s (locust.stats_logger:720)
[2021-07-09 08:39:27,587] [    INFO] - ---------------------------------------------------------------------------------------------
----------------------------------------------- (locust.stats_logger:722)
[2021-07-09 08:39:27,587] [    INFO] -  grpc query                                                     42737     0(0.00%)  |     316
      13    1050     310  |  234.60    0.00 (locust.stats_logger:725)
[2021-07-09 08:39:27,588] [    INFO] - ---------------------------------------------------------------------------------------------
----------------------------------------------- (locust.stats_logger:726)
[2021-07-09 08:39:27,588] [    INFO] -  Aggregated                                                     42737     0(0.00%)  |     316
      13    1050     310  |  234.60    0.00 (locust.stats_logger:727)
[2021-07-09 08:39:27,588] [    INFO] -  (locust.stats_logger:728)
[2021-07-09 08:39:27,612] [   ERROR] -
Addr [benchmark-dm9h7-1-milvus.qa-milvus.svc.cluster.local:19530] _execute_search_requests
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.DEADLINE_EXCEEDED
        details = "Deadline Exceeded"
        debug_error_string = "{"created":"@1625819967.545791044","description":"Error received from peer ipv4:10.96.127.22:19530","f
ile":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
        {'API start': '2021-07-09 08:34:27.545254', 'RPC start': '2021-07-09 08:34:27.545263', 'RPC error': '2021-07-09 08:39:27.611
948'} (pymilvus.client.grpc_handler:73)
[2021-07-09 08:39:27,612] [   ERROR] -
Addr [benchmark-dm9h7-1-milvus.qa-milvus.svc.cluster.local:19530] search
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.DEADLINE_EXCEEDED
        details = "Deadline Exceeded"
        debug_error_string = "{"created":"@1625819967.545791044","description":"Error received from peer ipv4:10.96.127.22:19530","f
ile":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
```

```
[2021-07-09 08:41:24,854] [    INFO] -  Name                                                          # reqs      # fails  |     Avg
     Min     Max  Median  |   req/s failures/s (locust.stats_logger:720)
[2021-07-09 08:41:24,855] [    INFO] - ---------------------------------------------------------------------------------------------
----------------------------------------------- (locust.stats_logger:722)
[2021-07-09 08:41:24,855] [    INFO] -  grpc query                                                     70084     0(0.00%)  |     570
      13  302631     330  |  219.30    0.00 (locust.stats_logger:725)
[2021-07-09 08:41:24,855] [    INFO] - ---------------------------------------------------------------------------------------------
----------------------------------------------- (locust.stats_logger:726)
[2021-07-09 08:41:24,855] [    INFO] -  Aggregated                                                     70084     0(0.00%)  |     570
      13  302631     330  |  219.30    0.00 (locust.stats_logger:727)
[2021-07-09 08:41:24,855] [    INFO] -  (locust.stats_logger:728)
[2021-07-09 08:41:24,856] [    INFO] - {'rps': 219.3, 'fail_ratio': 0.0, 'max_response_time': 302631, 'avg_response_time': 570.3} (m
ilvus_benchmark.main:96)
[2021-07-09 08:41:24,856] [   DEBUG] - {'type': 'locust_search_performance', 'value': {'rps': 219.3, 'fail_ratio': 0.0, 'max_respons
e_time': 302631, 'avg_response_time': 570.3}} (milvus_benchmark.main:104)
```
**Steps/Code to reproduce behavior**
https://argo-test.zilliz.cc/workflows/qa/benchmark-dm9h7?tab=workflow&nodeId=benchmark-dm9h7-2239452402&sidePanel=logs:benchmark-dm9h7-2239452402:main

logs: 
https://kibana-dev.zilliz.cc/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-1h,to:now))&_a=(columns:!(log),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:ad1e1c50-c9d9-11eb-9e1a-b3dcad9693fd,key:kubernetes.pod_name,negate:!f,params:!(benchmark-jtqnh-1-milvus-querynode-94cf475dc-v8s5q),type:phrases,value:benchmark-jtqnh-1-milvus-querynode-94cf475dc-v8s5q),query:(bool:(minimum_should_match:1,should:!((match_phrase:(kubernetes.pod_name:benchmark-jtqnh-1-milvus-querynode-94cf475dc-v8s5q))))))),hideChart:!t,index:ad1e1c50-c9d9-11eb-9e1a-b3dcad9693fd,interval:auto,query:(language:kuery,query:''),sort:!(!('@timestamp',desc)))

**Expected behavior**

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Software conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)
cluster

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

**Comment by xige-16** [comment]
NAME                                                   READY   STATUS    RESTARTS   AGE
benchmark-dm9h7-1-etcd-0                               1/1     Running   0          38s
benchmark-dm9h7-1-milvus-datacoord-7fccffb97d-g8zk5    1/1     Running   0          39s
benchmark-dm9h7-1-milvus-datanode-744b859975-2sjcq     1/1     Running   0          38s
benchmark-dm9h7-1-milvus-indexcoord-5744599f97-kcv4n   1/1     Running   0          38s
benchmark-dm9h7-1-milvus-indexnode-549d67b968-q4mqd    1/1     Running   0          38s
benchmark-dm9h7-1-milvus-proxy-858844cb57-7cmsz        1/1     Running   0          39s
benchmark-dm9h7-1-milvus-pulsar-758d7895db-t6j8v       1/1     Running   0          38s
benchmark-dm9h7-1-milvus-querycoord-59487bfb6-298xg    1/1     Running   0          39s
benchmark-dm9h7-1-milvus-querynode-d6ff86794-6t9qb     1/1     Running   0          39s
benchmark-dm9h7-1-milvus-querynode-d6ff86794-jfbcq     1/1     Running   0          39s
benchmark-dm9h7-1-milvus-querynode-d6ff86794-pxxsn     1/1     Running   0          39s
benchmark-dm9h7-1-milvus-rootcoord-5bff5c9fc8-rfj7s    1/1     Running   0          39s

**Comment by xige-16** [comment]
same with #6493

**Comment by xige-16** [comment]
@bigsheeper search stucked when querynode only load segment without watching DmChannels

**Comment by xige-16** [comment]
pr #6675 has fixed the issue.

**Comment by del-zhenwu** [comment]
still can be re-pro.
```
NAME                                                   READY   STATUS    RESTARTS   AGE
benchmark-d9c8r-1-etcd-0                               1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-datacoord-6b9db9c865-sq6qf    1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-datanode-567c977b8b-lwv7c     1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-indexcoord-667887b7bd-kcm2f   1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-indexnode-7ff57cf4c5-8nzl7    1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-proxy-76dd4d88c6-sj28h        1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-pulsar-7bc59c6fd4-25dhg       1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-querycoord-58d4dc54c-ggv44    1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-querynode-5fd5c64bd9-cw7z8    1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-querynode-5fd5c64bd9-n8rnd    1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-querynode-5fd5c64bd9-x2nf4    1/1     Running   0          9m48s
benchmark-d9c8r-1-milvus-rootcoord-6bf6ff56d9-nkv67    1/1     Running   0          9m48s
```
client logs:
https://argo-test.zilliz.cc/workflows/qa/benchmark-d9c8r?tab=workflow&nodeId=benchmark-d9c8r-3775476867&sidePanel=logs:benchmark-d9c8r-3775476867:main[2021-08-11 03:28:51,241] [    INFO] -  (locust.stats_logger:733)
[2021-08-11 03:28:51,242] [    INFO] - Ramping to 100 users at a rate of 2.00 per second (locust.runners:317)
[2021-08-11 03:28:51,242] [   DEBUG] - Ramping to {"MyUser": 2} (2 total users) (locust.runners:326)
[2021-08-11 03:28:51,243] [   DEBUG] - Spawning additional {"MyUser": 2} ({"MyUser": 0} already running)... (locust.runners:202)
[2021-08-11 03:28:51,244] [   DEBUG] - 2 users spawned (locust.runners:212)
[2021-08-11 03:28:51,244] [   DEBUG] - All users of class MyUser spawned (locust.runners:213)
[2021-08-11 03:28:51,245] [   DEBUG] - 0 users have been stopped, 2 still running (locust.runners:271)
[2021-08-11 03:29:21,263] [   ERROR] -
Addr [benchmark-d9c8r-1-milvus.qa-milvus.svc.cluster.local:19530] _execute_search_requests
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.DEADLINE_EXCEEDED
        details = "Deadline Exceeded"
        debug_error_string = "{"created":"@1628652561.262021163","description":"Error received from peer ipv4:10.96.205.166:19530","
file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
        {'API start': '2021-08-11 03:28:51.261178', 'RPC start': '2021-08-11 03:28:51.261189', 'RPC error': '2021-08-11 03:29:21.263
281'} (pymilvus.client.grpc_handler:76)
[2021-08-11 03:29:21,264] [   ERROR] -
Addr [benchmark-d9c8r-1-milvus.qa-milvus.svc.cluster.local:19530] search
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.DEADLINE_EXCEEDED
        details = "Deadline Exceeded"
        debug_error_string = "{"created":"@1628652561.262021163","description":"Error received from peer ipv4:10.96.205.166:19530","
file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
>
```

```

**Comment by xiaofan-luan** [comment]
@xige-16 is this same issue ebay hits?

**Comment by xige-16** [comment]
> @xige-16 is this same issue ebay hits?

yes

**Comment by xige-16** [comment]
The issue has fix by pr #7104, trouble you check it. @del-zhenwu

**Comment by del-zhenwu** [comment]
close
